plugins/rebalance/rest.go

```diff
@@ -0,0 +1,364 @@
+	GfDefragStatusLayoutFixStopped
+	// GfDefragStatusLayoutFixComplete should be set only for a volume that the rebalance fix-layout is completed
+	GfDefragStatusLayoutFixComplete
+	// GfDefragStatusLayoutFixFailed should be set only for a volume that are failed to run rebalance fix-layout
+	GfDefragStatusLayoutFixFailed
+)
+
+// RebalanceInfo represents Rebalance details
+type RebalanceInfo struct {
+	Volname           string
+	Status            RebalanceStatus
+	RebalanceID       uuid.UUID
+	RebalanceFiles    uint64
+	RebalanceData     uint64
+	LookedupFiles     uint64
+	RebalanceFailures uint64
+	ElapsedTime       uint64
+	SkippedFiles      uint64
+	TimeLeft          uint64
+	CommitHash        uint64
+}
+
+func createRebalanceInfo(rebalanceinfo *RebalanceInfo) (*RebalanceInfo, error) {
+	v := new(RebalanceInfo)
+	v.Volname = rebalanceinfo.Volname
+	v.RebalanceID = uuid.NewRandom()
+	v.Status = rebalanceinfo.Status
+	v.CommitHash = rebalanceinfo.CommitHash
+	return v, nil
+}
+
+func rebalanceStartInfoResp(v *RebalanceInfo) *api.RebalanceInfo {
+
+	return &api.RebalanceInfo{
+		Volname:     v.Volname,
+		Status:      api.RebalanceStatus(v.Status),
+		RebalanceID: v.RebalanceID,
+	}
+}
+
+func registerRebalanceStartStepFuncs() {
+	transaction.RegisterStepFunc(StartRebalance, "rebal-start.Commit")
+	transaction.RegisterStepFunc(storeRebalanceDetails, "rebal-start.StoreVolume")
+}
+
+func registerRebalanceStopStepFuncs() {
+	transaction.RegisterStepFunc(StopRebalance, "rebal-stop.Commit")
+}
+
+func registerRebalanceStatusStepFuncs() {
+	transaction.RegisterStepFunc(StatusRebalance, "rebal-status.Commit")
+}
+
+func rebalanceStart(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	rebalanceinfo := new(RebalanceInfo)
+
+	//Check volname given
+	if volname == "" {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not found", api.ErrCodeDefault)
+		return
+	}
+
+	//Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not started", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount == 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distributed volume or contain only 1 brick",
+			api.ErrCodeDefault)
+		return
+	}
+
+	/*Check volume is a tier volume or not
+	if vol.Type == volume.Tier {
+	        restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance operations are not supported on a tier volume",
+	                                api.ErrCodeDefault)
+	        return
+	}*/
+
+	// Check rebalance is already in process
+	if rebalanceinfo.Status == GfDefragStatusStarted {
```

Comment:
Wrong condition, rebalnceinfo is not fetched from store, it is initialized in Line 95